### PR TITLE
Examples and Abstract Classes

### PR DESCRIPTION
**Pull request by piraka9011**
Added abstract python and C++ class implementations for the BTAction to avoid redundancy. Makes writing the callback execution a bit easier.
Comes with examples.

### CODE REVIEW

File: behavior_tree_leaves/example_nodes/cpp/bt_action.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public:
+    BTAction(std::string name);
+    ~BTAction(void) {}
+    virtual void execute_callback(const behavior_tree_core::BTGoalConstPtr &goal) = 0;
+    void set_status(int status);    // Returns the status to the client (Behavior Tree)
+
+};
+
+BTAction::BTAction(std::string name):
+        as_(name, boost::bind(&BTAction::execute_callback, this, _1), false),
```
Comment (miccol) [inline behavior_tree_leaves/example_nodes/cpp/bt_action.h]: can it be replaced with `std::bind`? I would like to avoid boost dependencies

**Comment by miccol** [comment]
Looks awesome.
Check the boost::bind comment!

Thanks

**Comment by piraka9011** [comment]
No problem, switched `boost::bind` to `std::bind`.
I assumed that catkin automatically grabs boost libs but I guess this makes it more universal :)

**Comment by miccol** [comment]
yes you are right, but we cannot take for granted that catkin will grab boost also in the next version. Thanks

**Pull request merged**